File: src/Starfield2026.MiniToolbox/src/MiniToolbox.Garc/Textures/PocketMonsters/AD.cs
using System.IO;

using MiniToolbox.Garc.Containers;

namespace MiniToolbox.Garc.Textures.PocketMonsters
{
    class AD
    {
        /// <summary>
        ///     Loads all map textures (and other data) on a AD Pokémon container.
        /// </summary>
        /// <param name="data">The data</param>
        /// <returns>The Model group with textures and stuff</returns>
        public static RenderBase.OModelGroup load(Stream data)
        {
            RenderBase.OModelGroup models = new RenderBase.OModelGroup();

            OContainer container = PkmnContainer.load(data);
            for (int i = 1; i < container.content.Count; i++)
            {
                FileIO.LoadedFile file = FileIO.load(new MemoryStream(container.content[i].data));
                if (file.type == FileIO.formatType.model) models.merge((RenderBase.OModelGroup)file.data);
            }

            return models;
        }
    }
}

File: src/Starfield2026.BgEditor/backend/src/lib/ohana/Textures/PocketMonsters/AD.ts
import { BinaryReader } from '../../Core/BinaryReader.js';
import { PkmnContainer } from '../../Containers/PkmnContainer.js';
import { FileIO, formatType } from '../../Core/FileIO.js';
import { OModelGroup } from '../../Core/RenderBase.js';

/**
 * Loads all map textures (and other data) from an AD Pokemon container.
 * Ported from OhanaCli.Formats.Textures.PocketMonsters.AD (C#).
 */
export class AD {
  static load(input: BinaryReader): OModelGroup {
    const models = new OModelGroup();

    const container = PkmnContainer.load(input);
    // Note: starts at index 1 per original C# code
    for (let i = 1; i < container.content.length; i++) {
      const entry = container.content[i];
      if (!entry.data) continue;
      const file = FileIO.load(BinaryReader.fromBuffer(entry.data));
      if (file.type === formatType.model) {
        models.merge(file.data as OModelGroup);
      }
    }

    return models;
  }
}
